lib/dap/filter/simple.rb

```diff
@@ -171,6 +171,20 @@ def process(doc)
   end
 end
 
+class FilterFlatten
+  include Base
+  def process(doc)
+    self.opts.each_pair do |k,v|
+      if doc.has_key?(k) and doc[k].is_a?(Hash)
```

Comment:
Yeah, that works for me.  Also correct on hash vs arrays, as there'd have to be some sort of naming scheme for how to flatten an array in this context (foo.bar0, foo.bar1...).

Also, I had another thought -- instead of calling this flattening, how about calling it unnesting?